extensions/amp-story/1.0/amp-story-system-layer.js

```diff
@@ -282,18 +292,27 @@ export class SystemLayer {
   }
 
   /**
-   * @return {!Element}
    * @param {string} initialPageId
+   * @param {boolean=} isHidden
+   * @return {!Element}
    */
-  build(initialPageId) {
+  build(initialPageId, isHidden = false) {
     if (this.root_) {
       return this.root_;
     }
 
     this.systemLayerEl_ = renderSystemLayerElement(
       this.parentEl_,
-      this.progressBar_.build(initialPageId)
+      this.progressBar_.build(initialPageId),
+      isHidden
     );
+    if (isHidden) {
+      // The default value of `SYSTEM_UI_IS_VISIBLE_STATE` is `true`. We set it
+      // to `false` here solely to ensure that a subsequent firing of the
+      // `TOGGLE_SYSTEM_UI_IS_VISIBLE` action with a `true` value registers as a
+      // change in state instead of being a no-op.
+      this.storeService_.dispatch(Action.TOGGLE_SYSTEM_UI_IS_VISIBLE, false);
```

Comment:
(Plz see other comment, this should be set somewhere else and the system layer component should only consume the SYSTEM_UI_IS_VISIBLE state. You're making assumptions that system-layer is the only SYSTEM_UI which is not necessarily true, it should be set by a higher level component and consumed by all others)

Reply:
Makes sense, I've moved this dispatch to amp-story.js.